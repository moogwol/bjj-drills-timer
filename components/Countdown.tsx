import { View, StyleSheet } from "react-native";
import { Text, IconButton } from "react-native-paper";
import { useState, useEffect } from "react";
import { formattedTime } from "@/libraries/utility";


type CountdownProps = {
    seconds: string;
    minutes: string;
};



export default function Countdown(props: CountdownProps) {

    return (
        <View style={styles.buttonContainer} >
            <Text variant="displayLarge">{formattedTime(parseInt(props.seconds))}</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    buttonContainer: {
        flexDirection: "row",
        justifyContent: "space-around",
    },
});