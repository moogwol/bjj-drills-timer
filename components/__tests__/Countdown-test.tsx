import { render } from "@testing-library/react-native";
import Countdown from "../Countdown";

// Test the Countdown component renders correctly
describe('Countdown', () => {
    it('renders correctly', () => {
        const { toJSON } = render(
            <Countdown
                seconds="60"
                minutes="1"
            />
        );
        expect(toJSON()).toMatchSnapshot();
    });
});