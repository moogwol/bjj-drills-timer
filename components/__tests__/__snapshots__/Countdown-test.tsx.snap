// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 57,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 64,
          },
          undefined,
        ],
      ]
    }
  >
    01 : 00
  </Text>
</View>
`;
